test/powershell/Modules/Microsoft.PowerShell.Core/ModuleConstraint.Tests.ps1

```diff
@@ -0,0 +1,1514 @@
+    }
+
+    It "Does not load the module with FullyQualifiedName from the module path when ModuleVersion=<ModuleVersion>, MaximumVersion=<MaximumVersion>, RequiredVersion=<RequiredVersion>, Guid=<Guid>" -TestCases $guidFailCases {
+        param($ModuleVersion, $MaximumVersion, $RequiredVersion, $Guid)
+
+        $modSpec = New-ModuleSpecification -ModuleName $moduleName -ModuleVersion $ModuleVersion -MaximumVersion $MaximumVersion -RequiredVersion $RequiredVersion -Guid $Guid
+
+        { Import-Module -FullyQualifiedName $modSpec -ErrorAction Stop } | Should -Throw -ErrorId 'Modules_ModuleWithVersionNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand'
+    }
+
+    It "Does not load the module with FullyQualifiedName from the manifest when ModuleVersion=<ModuleVersion>, MaximumVersion=<MaximumVersion>, RequiredVersion=<RequiredVersion>, Guid=<Guid>" -TestCases $guidFailCases {
+        param($ModuleVersion, $MaximumVersion, $RequiredVersion, $Guid)
+
+        $modSpec = New-ModuleSpecification -ModuleName $manifestPath -ModuleVersion $ModuleVersion -MaximumVersion $MaximumVersion -RequiredVersion $RequiredVersion -Guid $Guid
+
+        { Import-Module -FullyQualifiedName $modSpec -ErrorAction Stop } | Should -Throw -ErrorId 'Modules_ModuleWithVersionNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand'
+    }
+
+    It "Does not load the module with version constraints from absolute path when ModuleVersion=<ModuleVersion>, MaximumVersion=<MaximumVersion>, RequiredVersion=<RequiredVersion>, Guid=<Guid>" -TestCases $failCases {
+        param($ModuleVersion, $MaximumVersion, $RequiredVersion, $Guid)
+
+        $sb = {
+            Invoke-ImportModule -Module $modulePath -MinimumVersion $ModuleVersion -MaximumVersion $MaximumVersion -RequiredVersion $RequiredVersion
+        }
+
+        if ($ModuleVersion -and $MaximumVersion -and ($ModuleVersion -ge $MaximumVersion))
+        {
+            $sb | Should -Throw -ErrorId 'ArgumentOutOfRange,Microsoft.PowerShell.Commands.ImportModuleCommand'
+            return
+        }
+        $sb | Should -Throw -ErrorId 'Modules_ModuleWithVersionNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand'
+    }
+
+    It "Does not load the module with version constraints from the module path when ModuleVersion=<ModuleVersion>, MaximumVersion=<MaximumVersion>, RequiredVersion=<RequiredVersion>, Guid=<Guid>" -TestCases $failCases {
+        param($ModuleVersion, $MaximumVersion, $RequiredVersion, $Guid)
+
+        $sb = {
+            Invoke-ImportModule -Module $modulePath -MinimumVersion $ModuleVersion -MaximumVersion $MaximumVersion -RequiredVersion $RequiredVersion
+        }
+
+        if ($ModuleVersion -and $MaximumVersion -and ($ModuleVersion -ge $MaximumVersion))
+        {
+            $sb | Should -Throw -ErrorId 'ArgumentOutOfRange,Microsoft.PowerShell.Commands.ImportModuleCommand'
```

Comment:
If all the negative test cases result in a terminating error, it may be simpler to have the FQErrorId as part of the testcase variation so this would simply be

> $sb | Should -Throw -ErrorId $ErrodId

Reply:
Just realised this second suggestion doesn't work just yet; `Import-Module -MaximumVersion -MinimumVersion` throws a different error to `Import-Module @{ ModuleVersion = ; MaximumVersion = }`.

That's what https://github.com/PowerShell/PowerShell/pull/7347 changes.

Reply:
Ok